### PR TITLE
Team reviews

### PR DESCRIPTION
**Pull request by Janis-Leuenberger**
https://redmine.renuo.ch/issues/16810
Screenshot 2021-11-18 at 14 47 53
Screenshot 2021-11-18 at 14 48 07
Screenshot 2021-11-18 at 14 48 16

### CODE REVIEW

File: __test__/integration/integration.test.ts [hunk 1]
```diff
@@ -103,26 +103,26 @@ describe('integration test', () => {
...
     it('has the correct subtitles', () => {
-      const subtitle = (index: number) => `renuo/github-pull-request-counter<b> #${index+1}</b>`;
+      const subtitle = (index: number) => `renuo/github-pull-request-counter<b> #${index + 1}</b>`;
 
-      expect(readProp('.link-container > .subdescription', 'innerHTML', 0)).resolves.toEqual(subtitle(0));
-      expect(readProp('.link-container > .subdescription', 'innerHTML', 1)).resolves.toEqual(subtitle(1));
-      expect(readProp('.link-container > .subdescription', 'innerHTML', 2)).resolves.toEqual(subtitle(2));
+      expect(readProp('.link-container > .subdescription', 'innerHTML', 0)).resolves.toContain(subtitle(0));
+      expect(readProp('.link-container > .subdescription', 'innerHTML', 1)).resolves.toContain(subtitle(1));
+      expect(readProp('.link-container > .subdescription', 'innerHTML', 2)).resolves.toContain(subtitle(2));
```
Comment (Janis-Leuenberger) [inline __test__/integration/integration.test.ts:125]: The `(x days ago)` feature should have broken this test already. Anyways using contains is a bit less work than trying to implement Timebot.

File: src/options.html [hunk 2]
```diff
@@ -41,26 +46,38 @@ <h2>Counter</h2>
...
-      <p>Enter a comma seperated list of GitHub account names from which the extension should fetch. (case insensitive)</p>
+      <p>Enter a comma seperated list of GitHub account names from which the extension should fetch. (case insensitive)
+      </p>
+    </div>
+    <input type="text" id="account-names" class="text-input" placeholder="User1, User2, User3, ..."></input>
+
+    <div class="heading-1">
+      <h2>Teams</h2>
+      <p>Enter a comma seperated list of GitHub teams of which the team review requests you would like to be displayed
+        seperatly. If a team is not present here it's review requests will be displayed in "In must review".
```
Comment (schmijos) [inline src/options.html:64]:
```suggestion
        seperatly. If a team is not present here it's review requests will be displayed in "I must review".
```

File: __test__/integration/integration.test.ts [hunk 3]
```diff
@@ -103,26 +103,26 @@ describe('integration test', () => {
     it('has the correct content', () => {
       expect(readProp('#link-to-renuo', 'href')).resolves.toEqual('https://www.renuo.ch/');
       expect(readProp('.title', 'innerHTML', 0)).resolves.toEqual('I must review');
-      expect(readProp('.title', 'innerHTML', 1)).resolves.toEqual('Someone must review');
+      expect(readProp('.title', 'innerHTML', 1)).resolves.toEqual('No review requested');
       expect(readProp('.title', 'innerHTML', 2)).resolves.toEqual('All reviews done');
       expect(readProp('.title', 'innerHTML', 3)).resolves.toEqual('Missing Assignee');
-      expect(readProp('.title', 'innerHTML', 4)).resolves.toEqual('Assigned');
+      expect(readProp('.title', 'innerHTML', 4)).resolves.toEqual('Assigned to me');
```
Comment (schmijos) [inline __test__/integration/integration.test.ts:109]: Shouldn't "Team must review" also be in this list?

File: src/js/services/github-api-wrapper.ts [hunk 4]
```diff
@@ -7,8 +7,27 @@ const GithubApiWrapper = async () => {
...
+    const teams = await SettingsStorageAccessor().loadTeams();
+    if (teams === '') return [];
+
+    let combinedPullRequests: Issue[] = [];
+
+    for (const team of teams.replace(/ /g, '').split(',')) {
+      const query = encodeURIComponent(`is:open is:pr team-review-requested:${team} archived:false`);
+      const pullRequests = await makeApiRequest('/search/issues', `q=${query}`);
+
+      /* istanbul ignore next */
```
Comment (schmijos) [inline src/js/services/github-api-wrapper.ts:26]: Not coverable by mock?

File: src/js/services/pull-request-storage-accessor.test.ts [hunk 5]
```diff
@@ -81,7 +82,14 @@ describe('StorageSerialzer', () => {
...
 
       it('loads empty arrays', () => {
-        expect(result).toEqual({ reviewRequested: [], noReviewRequested: [], allReviewsDone: [], missingAssignee: [], allAssigned: [] });
+        expect(result).toEqual({
+          reviewRequested: [],
+          teamReviewRequested: [],
+          noReviewRequested: [],
+          allReviewsDone: [],
+          missingAssignee: [],
+          allAssigned: [],
```
Comment (schmijos) [inline src/js/services/pull-request-storage-accessor.test.ts:91]: Maybe rename to `myAssigned`?

File: src/options.html [hunk 6]
```diff
@@ -18,9 +18,14 @@ <h2>Counter</h2>
       <label class="pointer" for="review-requested">I must review</label>
       <label class="hover-label">Someone requested a review from you!</label>
     </div>
+    <div class="hover-label-container">
+      <input class="pointer" type="checkbox" id="team-review-requested" name="team-review-requested">
+      <label class="pointer" for="team-review-requested">My team must review</label>
```
Comment (schmijos) [inline src/options.html:23]: Less to read:

```suggestion
      <label class="pointer" for="team-review-requested">Team reviews</label>
```

**Review by schmijos — APPROVED** [review]
Nice! Looking forward to the feature.

File: __test__/integration/integration.test.ts (hunk 3 above)
Comment (Janis-Leuenberger) [reply __test__/integration/integration.test.ts:109]: Yes 👍

File: src/js/services/github-api-wrapper.ts (hunk 4 above)
Comment (Janis-Leuenberger) [reply src/js/services/github-api-wrapper.ts]: It's quite hard to test... but I did it anyways now.

File: src/js/services/pull-request-storage-accessor.test.ts (hunk 5 above)
Comment (Janis-Leuenberger) [reply src/js/services/pull-request-storage-accessor.test.ts:91]: I'd rather not
Screenshot 2021-11-23 at 08 48 00

**Pull request merged**